include/boost/json/impl/serializer.ipp

```diff
@@ -675,44 +680,69 @@ write_value(stream& ss)
         default:
         case state::nul1: case state::nul2:
         case state::nul3: case state::nul4:
-            return write_null<StackEmpty>(ss);
+            return write_null<StackEmpty>(ss, opt);
 
         case state::tru1: case state::tru2:
         case state::tru3: case state::tru4:
-            return write_true<StackEmpty>(ss);
+            return write_true<StackEmpty>(ss, opt);
 
         case state::fal1: case state::fal2:
         case state::fal3: case state::fal4:
         case state::fal5:
-            return write_false<StackEmpty>(ss);
+            return write_false<StackEmpty>(ss, opt);
 
         case state::str1: case state::str2:
         case state::str3: case state::str4:
         case state::esc1:
         case state::utf1: case state::utf2:
         case state::utf3: case state::utf4:
         case state::utf5:
-            return write_string<StackEmpty>(ss);
+            return write_string<StackEmpty>(ss, opt);
 
         case state::num:
-            return write_number<StackEmpty>(ss);
+            return write_number<StackEmpty>(ss, opt);
 
         case state::arr1: case state::arr2:
         case state::arr3: case state::arr4:
-            return write_array<StackEmpty>(ss);
+            return write_array<StackEmpty>(ss, opt);
 
         case state::obj1: case state::obj2:
         case state::obj3: case state::obj4:
         case state::obj5: case state::obj6:
-            return write_object<StackEmpty>(ss);
+            return write_object<StackEmpty>(ss, opt);
         }
     }
 }
 
 string_view
 serializer::
 read_some(
-    char* dest, std::size_t size)
+    char* dest, std::size_t size, const serialize_options & opt)
```

Comment:
Why are we passing the options into every call to `read_some`? Shouldn't we construct with the options instead, and then access the options as a data member instead of passing it as a parameter to every member function?